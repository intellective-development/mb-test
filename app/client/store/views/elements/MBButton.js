// @flow

import * as React from 'react';
import bindClassNames from '../../../shared/utils/bind_classnames';

import styles from './MBButton.scss';

const cn = bindClassNames(styles);

// This module exports a wrapper around the basic html button element.
// The idea is to provide a new baseline for us to work off of as we move towards component style sharing.
// Eventually, the goal is to fit this component and its props into the styleguide the design team is working towards.

type MBButtonProps = {
  className?: string,
  type?: 'default' | 'hollow' | 'action',
  size?: 'medium' | 'small' | 'tall', // TODO: transition to a boolean, small or not small
  expand?: boolean
};

const MBButton = ({className, type = 'default', size = 'medium', expand = false, ...rest_props}: MBButtonProps) => {
  const classes = cn('elMBButton', {
    // types
    elMBButton__Default: type === 'default',
    elMBButton__Hollow: type === 'hollow',
    elMBButton__Action: type === 'action',

    // sizes
    elMBButton__Medium: size === 'medium',
    elMBButton__Small: size === 'small',
    elMBButton__Tall: size === 'tall',

    elMBButton__Expand: expand
  }, className);

  return <button {...rest_props} className={classes} />;
};

export default MBButton;
