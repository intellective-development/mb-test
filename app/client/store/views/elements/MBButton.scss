@import "../style/variables";

// this is still somewhat relying on the global button styles
// prefixed with button to help with some stuff
.elMBButton {
  outline: none; // turn off default system interaction styles
  // from .button
  -webkit-appearance: none;
  border: 1px solid $brand-black;
  border-radius: 4px;
  color: $brand-white;
  cursor: pointer;
  font-family: $font-family-avenir;
  font-weight: $heavy-font-weight;
  line-height: 1.3em;
  margin: 0;
  padding-bottom: 0.75em;
  padding-top: 0.875em;
  text-transform: uppercase;
  transition: background-color 300ms ease-out, border-color 300ms ease-out;

  &:focus {
    color: initial;
    background-color: initial;
  }
}

.elMBButton__Default {
  @include pixel-antialiased;
  @include button-disabled-color(
    $brand-black, // unchanged
    $brand-black, // unchanged
    $brand-white, // unchanged
    0.8
  );

  background-color: $brand-black;
  border-color: $brand-black;
  color: $brand-white;

  &:hover,
  &:active {
    background-color: lighten($brand-black, 20%);
    border-color: lighten($brand-black, 20%);
  }
}

.elMBButton__Hollow {
  @include button-disabled-color(
    $brand-transparent, // unchanged
    desaturate($brand-black, 40%),
    desaturate($brand-black, 40%)
  );
  background-color: $brand-transparent;
  border-color: $brand-black;
  color: $brand-black;

  &:hover,
  &:active {
    background-color: $brand-pale-blue;
    color: $brand-black;
  }
}

.elMBButton__Action {
  @include pixel-antialiased;
  @include button-disabled-color(
    desaturate($brand-red, 70%),
    desaturate($brand-red, 70%),
    $brand-white // unchanged
  );

  background: $brand-red;
  border-color: $brand-red;
  color: $brand-white;

  &:hover,
  &:active {
    background-color: lighten($brand-red, 10%);
    border-color: lighten($brand-red, 10%);
  }
}

// TODO: consolidate medium/tall, they're the default size
.elMBButton__Medium {
  @include font-size(14);

  letter-spacing: 0.125rem;
}

.elMBButton__Tall {
  @include font-size(14);

  letter-spacing: 0.125rem;
  min-height: 48px;
}

.elMBButton__Small {
  @include font-size(13);

  letter-spacing: 0;

  // goal is 32px tall
  min-height: 32px;
  padding: 0.6em 1.125em 0.35em;
  text-transform: capitalize;
}

.elMBButton__Expand {
  width: 100%;
}
