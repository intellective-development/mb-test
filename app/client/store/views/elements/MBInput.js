// @flow
import * as React from 'react';
import bindClassNames from '../../../shared/utils/bind_classnames';

import styles from './MBInput.scss';

const cn = bindClassNames(styles);

// This module exports a wrapper around the basic html input element.
// The idea is to provide a new baseline for us to work off of as we move towards component style sharing.
// Eventually, the goal is to fit this component and its props into the styleguide the design team is working towards.

type MBInputProps = {className: string, inputRef: Function} & React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>>;
export const Input = ({className, inputRef, ...input_props}: MBInputProps) => {
  const classes = cn('elMBInput', className);
  return <input {...input_props} className={classes} ref={inputRef} />;
};
