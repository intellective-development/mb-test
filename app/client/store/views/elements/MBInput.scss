@import "../style/variables";

// TODO: there are a lot of styles on the input element at the moment, we'd like to move them in here
// For now, I'm just adding in styles we should be using going forward
input.elMBInput {
  font-family: $font-family-avenir;
  border-radius: 4px;
  height: 36px; // TODO: consider just resetting this to the initial height
  margin: initial;
  // height: initial; // This enables either instances or their parents to specify the size of the input

  padding-left: 0.4em;
  padding-right: 0.4em;

  &:focus {
    outline: 0;
  }
}
