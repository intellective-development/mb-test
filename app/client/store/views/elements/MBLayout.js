// @flow

import * as React from 'react';
import cn from 'classnames';

// This module exports a set of small layouts we use in multiple places in the app. At the moment, it's a bit of a catch all
// The idea is to provide a new baseline for us to work off of as we move towards component style sharing.
// Eventually, the goal is to fit this component and its props into the styleguide the design team is working towards.

type ButtonInputProps = {className?: string, children?: React.Node};
export const ButtonInput = ({className, children}: ButtonInputProps) => {
  const classes = cn('el-mblayouts-bi__container', className);
  return (
    <div className={classes}>
      {children}
    </div>
  );
};

type ButtonGroupProps = {className?: string, children?: React.Node};
export const ButtonGroup = ({className, children}: ButtonGroupProps) => {
  const classes = cn('el-mblayouts-bg__container', className);
  return (
    <div className={classes}>
      {children}
    </div>
  );
};

type StandardGridProps = {className?: string, no_padding?: boolean, children?: React.Node};
export const StandardGrid = ({className, no_padding, children, ...rest_props}: StandardGridProps) => {
  const classes = cn('el-mblayouts-sg', {_pad_sides: !no_padding}, className);

  return (
    <div {...rest_props} className={classes}>
      {children}
    </div>
  );
};
