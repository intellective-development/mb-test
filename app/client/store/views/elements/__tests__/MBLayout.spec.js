
import * as React from 'react';

// import TestProvider from 'client/store/__tests__/utils/TestProvider';
import MBButton from '../MBButton';
import * as MBInput from '../MBInput';
import * as MBLayout from '../MBLayout';

describe('MBLayout.ButtonInput', () => {
  it('renders', () => {
    expect(render(
      <MBLayout.ButtonInput>
        <MBInput.Input />
        <MBButton size="small">Go</MBButton>
      </MBLayout.ButtonInput>
    )).toMatchSnapshot();
  });
});

describe('MBLayout.ButtonGroup', () => {
  it('renders', () => {
    expect(render(
      <MBLayout.ButtonGroup>
        <MBButton size="small">Confirm</MBButton>
        <MBButton size="small">Deny</MBButton>
      </MBLayout.ButtonGroup>
    )).toMatchSnapshot();
  });
});
