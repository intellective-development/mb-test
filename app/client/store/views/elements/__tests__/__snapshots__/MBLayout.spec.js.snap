// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBLayout.ButtonGroup renders 1`] = `
<div
  class="el-mblayouts-bg__container"
>
  <button
    class="elMBButton elMBButton__Default elMBButton__Small"
  >
    Confirm
  </button>
  <button
    class="elMBButton elMBButton__Default elMBButton__Small"
  >
    Deny
  </button>
</div>
`;

exports[`MBLayout.ButtonInput renders 1`] = `
<div
  class="el-mblayouts-bi__container"
>
  <input
    class="elMBInput"
  />
  <button
    class="elMBButton elMBButton__Default elMBButton__Small"
  >
    Go
  </button>
</div>
`;
